client/state/data-layer/wpcom/sites/gutenberg/index.js

```diff
@@ -30,9 +30,12 @@ const fetchGutenbergOptInData = action =>
 		action
 	);
 
-const setGutenbergOptInData = ( { siteId }, { editor_web: editor, opt_in: optIn } ) => dispatch => {
+const setGutenbergOptInData = (
+	{ siteId },
+	{ editor_web: editor, opt_in: optIn, opt_out: optOut }
+) => dispatch => {
 	dispatch( { type: EDITOR_TYPE_SET, siteId, editor } );
-	dispatch( { type: GUTENBERG_OPT_IN_SET, siteId, optIn } );
+	dispatch( { type: GUTENBERG_OPT_IN_SET, siteId, optIn: optIn || optOut } );
```

Comment:
This confuses me. `GUTENBERG_OPT_IN_SET` is intended to set in the state the value of `opt_in`. This could cause an inconsistency when `opt_in` has a falsy value and `opt_out` has a truthy value, since `isGutenbergOptInEnabled` will return `true` on those cases.

Why don't we store in a separate property the `opt_out` value?

Reply:
Let's do it! I verified that this code does not change existing behavior (because we only ever check for opt_in = null if classic, and opt_out = null if gutenberg, so the existing logic does "work"). It also confuses me and this is definitely the right change :)